pkg/continuousprofiling/profiler.go

```diff
@@ -35,7 +37,7 @@ var (
 // DefaultConfig creates a new configuration with default properties.
 func DefaultConfig() *pyroscope.Config {
 	return &pyroscope.Config{
-		ApplicationName:   env.ContinuousProfilingAppName.Setting(),
+		ApplicationName:   utils.IfThenElse[string](env.ContinuousProfilingAppName.Setting() != "", env.ContinuousProfilingAppName.Setting(), os.Getenv("POD_NAME")),
```

Comment:
Have you considered using `WithDefault` with `ContinuousProfilingAppName` instead of the fallback here?

Reply:
What I'm suggesting is something like
```
ContinuousProfilingAppName = RegisterSetting("ROX_CONTINUOUS_PROFILING_APP_NAME", WithDefault(os.Getenv("POD_NAME")))
```

Reply:
Ah! That makes a lot of sense! For some reason I thought you meant `WithDefaultAppName` 😅 I'll change the code to do what you suggested.